// sample-project/src/sample-screen1/index.tsx
import * as React from 'react';
import * as WFace from '@wface/components';

export interface SampleScreen1State {
    name?: string,
    surname?: string
}

export class SampleScreen1 extends React.Component<WFace.BaseScreenProps, SampleScreen1State> {
    constructor(props: WFace.BaseScreenProps) {
        super(props);

        this.state = this.props.screenContext.state || {
            name: 'mehmet',
            surname: 'baran'
        }
    }

    public changeValues() {
        this.setState({
            name: 'x',
            surname: 'y'
        })
    }

    public render() {
        console.log('render' + this.state.name + this.state.surname);
        return (        
            <WFace.WTypography>
                SampleScreen1 works                
                <br/>
                <WFace.WTextField value={this.state.name}/>
                <input value={this.state.name} />
                <br/>
                <WFace.WTextField value={this.state.surname}/>
                <br/>
                <WFace.WButton onClick={() => this.changeValues() }>Deneme</WFace.WButton>
            </WFace.WTypography>
        );
    }
}
